finch/processes/wps_sdba.py

```diff
@@ -0,0 +1,220 @@
+        "netCDF output",
+        as_reference=True,
+        supported_formats=[FORMATS.NETCDF],
+    ),
+    wpsio.output_log,
+    wpsio.output_metalink,
+]
+
+
+class EmpiricalQuantileMappingProcess(FinchProcess):
+    def __init__(self):
+        inputs = (
+            list(resources.values())
+            + list(group_args.values())
+            + list(adjust_args.values())
+            + [
+                wpsio.variable_any,
+                LiteralInput(
+                    "nquantiles",
+                    "Number of quantile bins",
+                    abstract="The number of quantiles to use. Two endpoints at 1e-6 and 1 - 1e-6 will be added.",
+                    data_type="integer",
+                    default=20,
+                    min_occurs=0,
+                ),
+                LiteralInput(
+                    "kind",
+                    "Kind of adjustment (+, *)",
+                    abstract="Use * for multiplicative adjustment, or + for additive adjustement.",
+                    data_type="string",
+                    default="+",
+                    allowed_values=["+", "*"],
+                    min_occurs=0,
+                ),
+            ]
+        )
+
+        super().__init__(
+            self._handler,
+            identifier="empirical_quantile_mapping",
+            title="Empirical Quantile Mapping bias-adjustment",
+            version="0.1",
+            abstract="Adjustment factors are computed between the quantiles of `ref` and `sim`."
+            "Values of `sim` are matched to the corresponding quantiles of `hist` and corrected accordingly.",
+            inputs=inputs,
+            outputs=common_outputs,
+            status_supported=True,
+            store_supported=True,
+        )
+
+    def _handler(self, request, response):
+        def _log(message, percentage):
+            write_log(self, message, subtask_percentage=percentage)
+
+        res = {}
+        group = {}
+        train = {}
+        adj = {}
+
+        try:
+            variable = request.inputs.pop(wpsio.variable.identifier, None)
+
+            for key, input in request.inputs.items():
+                if key in resources:
+                    ds = try_opendap(request.inputs[key][0])
+                    name = variable or list(ds.data_vars)[0]
+                    res[key] = ds[name]
+
+                elif key in group_args:
+                    group[key] = single_input_or_none(request.inputs, key)
+
+                elif key in adjust_args:
+                    adj[key] = single_input_or_none(request.inputs, key)
+
+                else:
+                    train[key] = single_input_or_none(request.inputs, key)
+
+            _log("Successfully read inputs from request.", 1)
+
+            group = xclim.sdba.Grouper(**group)
+            _log("Grouper object created.", 2)
+
+            bc = xclim.sdba.EmpiricalQuantileMapping(**train, group=group)
+            bc.train(res["ref"], res["hist"])
```

Comment:
I suggest we convert ref, hist and sim to a no_leap calendar by default before running the adjustment. This could also be controlled by an argument, but it doesn't make much sense to not do it anyway.

Reply:
ok. Please see commit to confirm I got it right.

Reply:
Complicates testing though...